Iterable/iterable-web-sdk
# [MOB-7613] Remove extra height from iframe + refactor

## JIRA Ticket(s) if any

* [MOB-7613](https://iterable.atlassian.net/browse/MOB-7613)

## Description

Remove extra height added to iframe style. Originally addressed styling issue with WYSIWYG templates, but is handled better now by simply overriding the browser's user agent stylesheet on the body element itself.

Set the iframe height on load, because the image preloader is not a foolproof way to ensure all images get loaded in advance (I tested this to be the case for <img> tags).

Refactor cache-related utils to separate file.
Refactor types.

## Test Steps

Given I send myself templates made with BEE and WYSIWYG editors
When I render the messages on my browser
Then I see no unexpected margin around the message
And I see the iframe height is automatically adjusted based on all images loaded in the message

Regression testing on general functionality, esp. for...
* display position (center, top right, bottom right, full)
* display option (immediate vs deferred)
* close button position (top right, top left)
* link handling (open new tab, open same tab, external new tab)
...because typings have been refactored for these.

[MOB-7613]: https://iterable.atlassian.net/browse/MOB-7613?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
